src/cascadia/TerminalApp/ActionArgs.h

```diff
@@ -243,28 +245,99 @@ namespace winrt::TerminalApp::implementation
+    static constexpr std::string_view CellString{ "cell" };
+    static constexpr std::string_view WordString{ "word" };
+    static constexpr std::string_view ViewportString{ "viewport" };
+    static constexpr std::string_view BufferString{ "buffer" };
+
+    // Function Description:
+    // - Helper function for parsing a SelectionExpansionMode from a string
+    // Arguments:
+    // - directionString: the string to attempt to parse
+    // Return Value:
+    // - The encoded Direction value, or Direction::None if it was an invalid string
+    static winrt::Microsoft::Terminal::Settings::SelectionExpansionMode ParseExpansionMode(const std::string& expansionModeString)
+    {
+        if (expansionModeString == CellString)
+        {
+            return winrt::Microsoft::Terminal::Settings::SelectionExpansionMode::Cell;
+        }
+        else if (expansionModeString == WordString)
+        {
+            return winrt::Microsoft::Terminal::Settings::SelectionExpansionMode::Word;
+        }
+        else if (expansionModeString == ViewportString)
+        {
+            return winrt::Microsoft::Terminal::Settings::SelectionExpansionMode::Viewport;
+        }
+        else if (expansionModeString == BufferString)
+        {
+            return winrt::Microsoft::Terminal::Settings::SelectionExpansionMode::Buffer;
+        }
+        // default behavior for invalid data
+        return winrt::Microsoft::Terminal::Settings::SelectionExpansionMode::Cell;
+    };
+
+    struct MoveSelectionAnchorArgs : public MoveSelectionAnchorArgsT<MoveSelectionAnchorArgs>
+    {
+        MoveSelectionAnchorArgs() = default;
+        GETSET_PROPERTY(winrt::Microsoft::Terminal::Settings::Direction, Direction, winrt::Microsoft::Terminal::Settings::Direction::None);
+        GETSET_PROPERTY(winrt::Microsoft::Terminal::Settings::SelectionExpansionMode, ExpansionMode, winrt::Microsoft::Terminal::Settings::SelectionExpansionMode::Cell);
+
+        static constexpr std::string_view DirectionKey{ "direction" };
+        static constexpr std::string_view ExpansionModeKey{ "expansionMode" };
+
+    public:
+        bool Equals(const IActionArgs& other)
+        {
+            auto otherAsUs = other.try_as<MoveSelectionAnchorArgs>();
+            if (otherAsUs)
+            {
+                return (otherAsUs->_Direction == _Direction) && (otherAsUs->_ExpansionMode == _ExpansionMode);
+            }
+            return false;
+        };
+        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        {
+            // LOAD BEARING: Not using make_self here _will_ break you in the future!
+            auto args = winrt::make_self<MoveSelectionAnchorArgs>();
+            if (auto directionString{ json[JsonKey(DirectionKey)] })
+            {
+                args->_Direction = ParseDirection(directionString.asString());
+
+                if (auto expansionModeString{ json[JsonKey(ExpansionModeKey)] })

```

Comment:
Should we only be parsing an expansion mode if there's a direction? Or should we always parse the expansion mode?

Reply:
I think `direction` should be required because it doesn't really make sense to assume a default `direction`. That said, not entirely sure if that's being enforced...

At the very least I should make sure the schema/doc says that (which I have yet to update -.-)